notebooks/Testgrid_flakiness_detection.ipynb

```diff
@@ -0,0 +1,1458 @@
+{
```

Comment:
<p>Are there any other flake analyses results we can compare this output to? maybe the current testgrid flake labels? Are there any claims we can make here about how well this approach performs compared to another? Even the naive approach above?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aicoe-aiops/ocp-ci-analysis/pull/37/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='33'/>

Reply:
<p>There are some basic test cases to verify the results of flip analyzer. I compared the results. It's consistent with it. Link of test cases: https://github.com/GoogleCloudPlatform/testgrid/blob/2a016ff75a27fff1550f5e9f5319cf0114182547/pkg/summarizer/analyzers/flipanalyzer_test.go</p><p> </p><p>The only difference is I multiplied the final output by 2. There is a small bug in their implementation, and the score will never go above 50. If the test is perfect flip with [p, f, p, f, p, f, p, f, p, f] behavior, it's giving output as 50, and there is no way it can go higher than 50. Also, they didnt consider not run the tests into flip flakiness calculation.</p><p> </p><p>I just added test cases for the comparison of the naive method vs. flip flakiness method and the flip flakiness method vs. flip flakiness method with optimal distance.&nbsp;</p>

Reply:
<p>Current, dashboard didn't give a flakiness score but  trigger a flakiness alert if score is greater than 1. </p><p>You can check it here: https://github.com/GoogleCloudPlatform/testgrid/blob/a18fe953cf98174c215c43e0258b0515e37c283b/pb/config/config.proto#L621</p>

Reply:
>There are some basic test cases to verify the results of flip analyzer. I compared the results. It's consistent with it. Link of test cases

Are these tests results included in the notebook?

Reply:
> I just added test cases for the comparison of the naive method vs. flip flakiness method and the flip flakiness method vs. flip flakiness method with optimal distance. 

The results seem to be a bit hidden in your comments, it would be nice if there was a section at the end that compared results for all the methods against your designed examples as well as against the entire testgrid your using.

Reply:
>Hence, it's not best way to comparison

What do you think would be a good way to make the comparison? Since testgrid already has a mechanism for detecting flakes at the test level, then aggregates it up to a flakiness alert for the entire grid, would doing the same thing here possibly provide a clearer comparison? WDYT? 

I think my larger point here is, if we are trying to provide value to testgrid and we are doing it through an improved approach to flake detection, we need to have a way to compare the existing flake detection mechanism and ours, right?

Reply:
> What do you think would be a good way to make the comparison? Since testgrid already has a mechanism for detecting flakes at the test level, then aggregates it up to a flakiness alert for the entire grid, would doing the same thing here possibly provide a clearer comparison? WDYT?

It makes sense. 

> I think my larger point here is, if we are trying to provide value to testgrid and we are doing it through an improved approach to flake detection, we need to have a way to compare the existing flake detection mechanism and ours, right?

Yeah, I am on it. :smile:

Reply:
> Are these tests results included in the notebook?

Yes, I added results of those test cases.

Reply:
> The results seem to be a bit hidden in your comments, it would be nice if there was a section at the end that compared results for all the methods against your designed examples as well as against the entire testgrid your using.

Done, I added a separate section.

Reply:
> What do you think would be a good way to make the comparison? Since testgrid already has a mechanism for detecting flakes at the test level, then aggregates it up to a flakiness alert for the entire grid, would doing the same thing here possibly provide a clearer comparison? WDYT?
> 
> I think my larger point here is, if we are trying to provide value to testgrid and we are doing it through an improved approach to flake detection, we need to have a way to compare the existing flake detection mechanism and ours, right?

I generated flakey labels given by test grid and also generated labels by our method. We can compare the test grid visualization of both methods.